File: chudnovsky/src/main/java/com/apps/elliotgrin/chudnovsky/Range.java
package com.apps.elliotgrin.chudnovsky;

public class Range {
    public long initalK;
    public long finalK;

    public Range(long initalK, long finalK) {
        if (initalK < 0 || finalK < 0) {
            throw new IllegalArgumentException("Bounds should be strictly positive.");
        }

        if (finalK < initalK) {
            throw new IllegalArgumentException("Upper bound should be greater than lower bound.");
        }

        this.initalK = initalK;
        this.finalK = finalK;
    }

    @Override
    public String toString() {
        return initalK + " " + finalK;
    }
}

File: chudnovsky/src/main/java/com/apps/elliotgrin/chudnovsky/Range.kt
package com.apps.elliotgrin.chudnovsky

class Range(val initialK: Long, val finalK: Long) {

    init {
        if (initialK < 0 || finalK < 0) {
            throw IllegalArgumentException("Bounds should be strictly positive.")
        }

        if (finalK < initialK) {
            throw IllegalArgumentException("Upper bound should be greater than lower bound.")
        }
    }

    override fun toString(): String {
        return initialK.toString() + " " + finalK
    }
}
